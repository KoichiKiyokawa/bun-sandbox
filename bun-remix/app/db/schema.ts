import { sql } from "drizzle-orm";
import { integer, sqliteTable, text } from "drizzle-orm/sqlite-core";

export const usersSchema = sqliteTable("users", {
  id: text("id").primaryKey().$defaultFn(crypto.randomUUID),
  name: text("name").notNull(),
  email: text("email").notNull().unique(),
  createdAt: text("created_at")
    .notNull()
    .default(sql`DATETIME('now')`),
  updatedAt: text("updated_at")
    .notNull()
    .default(sql`DATETIME('now')`),
});

export const todosSchema = sqliteTable("todos", {
  id: text("id").primaryKey().$defaultFn(crypto.randomUUID),
  text: text("text").notNull(),
  completed: integer("completed", { mode: "boolean" }).notNull(),
});
