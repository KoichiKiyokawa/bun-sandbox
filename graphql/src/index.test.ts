import { expect, test } from "bun:test";
import { createYoga } from "graphql-yoga";
import { schema } from "./schema";

const yoga = createYoga({ schema });

test("hello query", async () => {
  const res = await yoga.fetch("http://localhost:3000/graphql", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ query: "{ hello }" }),
  });
  const json = await res.json();

  expect(json).toMatchObject({ data: { hello: "Hello from Yoga!" } });
});
