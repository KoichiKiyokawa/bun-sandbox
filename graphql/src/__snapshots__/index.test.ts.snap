// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`hello query 1`] = `
{
  "data": {
    "hello": "Hello from Yoga!",
  },
}
`;
